### PR TITLE
[PyTorch] Add test for all-masked case for native softmax

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75809
* #75808
* __->__ #75803
* #75347
* #75346

It returns all NaNs. CUDA implementation required a fix for this.

Differential Revision: [D35327730](https://our.internmc.facebook.com/intern/diff/D35327730/)

### CODE REVIEW

**Comment by swolchok** [comment]
previously commented on as https://github.com/pytorch/pytorch/pull/75493, had to send a new PR due to tooling issues

**Review by ngimel — APPROVED** [review]

**Comment by cpuhrsch** [comment]
@swolchok - please start adding these labels, we all do it

**Comment by swolchok** [comment]
I do add them, but I don't check github notifications particularly often and you keep beating me to it :)
